system/lib/wasmfs/wasmfs.cpp

```diff
@@ -0,0 +1,39 @@
+// Copyright 2021 The Emscripten Authors.  All rights reserved.
+// Emscripten is available under two separate licenses, the MIT license and the
+// University of Illinois/NCSA Open Source License.  Both these licenses can be
+// found in the LICENSE file.
+// This file defines the global state of the new file system.
+// Current Status: Work in Progress.
+// See https://github.com/emscripten-core/emscripten/issues/15041.
+
+#include "wasmfs.h"
+#include "streams.h"
+
+namespace wasmfs {
+
+// The below lines are included to make the preprocessor believe that the global
+// constructor is included as a header. This ensures that the global state of
+// the file system is constructed before anything else. ATTENTION: No other
+// static global objects should be defined besides wasmFS.
+// Due to # define _LIBCPP_INIT_PRIORITY_MAX
+// __attribute__((init_priority(101))), we must use init priority 100 (reserved
+// system priority) since wasmFS is a system level component.
```

Comment:
Maybe can add a TODO here
```suggestion
// system priority) since wasmFS is a system level component.
// TODO: consider instead adding this in libc's startup code.
```